src/matter/cluster/NetworkCommissioningCluster.ts

```diff
@@ -4,25 +4,241 @@
+    TlvBoolean,
+    TlvByteString,
+    TlvEnum,
+    TlvField,
+    TlvInt32,
+    TlvObject,
+    TlvUInt8,
+    TlvNullable,
+    TlvUInt16,
+    TlvInt8,
+    TlvBitmap,
+    TlvUInt64,
+    TlvOptionalField,
+    TlvString
+} from "@project-chip/matter.js";
+import {
+    Attribute,
+    Cluster,
+    Command,
+    OptionalAttribute,
+    OptionalCommand,
+    WritableAttribute
+} from "./Cluster";
 
+/** @see {@link MatterApplicationClusterSpecificationV1_0} § 11.8.6.5 */
 export const enum NetworkCommissioningStatus {
+    /** OK, no error. */
     Success = 0,
+    /** Value Outside Range. */
     OutOfRange = 1,
+
+    /** A collection would exceed its size limit. */
     BoundsExceeded = 2,
+
+    /** The NetworkID is not among the collection of added networks. */
     NetworkIDNotFound = 3,
+
+    /** The NetworkID is already among the collection of added networks. */
     DuplicateNetworkID = 4,
+
+    /** Cannot find AP: SSID Not found. */
     NetworkNotFound = 5,
+
+    /** Cannot find AP: Mismatch on band/channels/regulatory domain / 2.4GHz vs 5GHz. */
     RegulatoryError = 6,
+
+    /** Cannot associate due to authentication failure. */
     AuthFailure = 7,
+
+    /** Cannot associate due to unsupported security mode. */
     UnsupportedSecurity = 8,
+
+    /** Other association failure. */
     OtherConnectionFailure = 9,
+
+    /** Failure to generate an IPv6 address. */
     IPV6Failed = 10,
+
+    /** Failure to bind Wi-Fi <-> IP interfaces. */
     IPBindFailed = 11,
+
+    /** Unknown error. */
     UnknownError = 12,
 }
 
+/** @see {@link MatterApplicationClusterSpecificationV1_0} § 11.8.6.2 */
+export const enum WiFiBand {
+    /** 2.4GHz - 2.401GHz to 2.495GHz (802.11b/g/n/ax) */
+    '2G4' = 0,
+    /** 3.65GHz - 3.655GHz to 3.695GHz (802.11y) */
+    '3G65' = 1,
+    /** 5GHz - 5.150GHz to 5.895GHz (802.11a/n/ac/ax) */
+    '5G' = 2,
+    /** 6GHz - 5.925GHz to 7.125GHz (802.11ax / WiFi 6E) */
+    '6G' = 3,
+    /** 60GHz - 57.24GHz to 70.20GHz (802.11ad/ay) */
+    '60G' = 4,
+}
+
+/** @see {@link MatterApplicationClusterSpecificationV1_0} § 11.8.6.1 */
+export const TlvWiFiSecurity = TlvBitmap(TlvUInt8, {
+    Unencrypted: BitFlag(0),
+    Wep: BitFlag(1),
+    'WPA-PERSONAL': BitFlag(2),
+    'WPA2-PERSONAL': BitFlag(3),
+    'WPA3-PERSONAL': BitFlag(4),
+});
+
+/** @see {@link MatterCoreSpecificationV1_0} § 11.8.6.4 */
+const TlvWiFiInterfaceScanResult = TlvObject({
+    security: TlvField(0, TlvWiFiSecurity),
+    ssid: TlvField(1, TlvByteString.bound({ maxLength: 32 })),
+    bssid: TlvField(2, TlvByteString.bound({ length: 6 })),
+    channel: TlvField(3, TlvUInt16),
+
+    /** Used to differentiate overlapping channel number values across different Wi-Fi frequency bands. */
+    wiFiBand: TlvOptionalField(4, TlvEnum<WiFiBand>()),
+
+    /** Denote the signal strength in dBm of the associated scan result. */
+    rssi: TlvOptionalField(5, TlvInt8),
+});
+
+// TODO: Move to TlvString.ts in matter.js!
```

Comment:
Can you give more details about this TODO?

Reply:
<img width="968" alt="Bildschirm­foto 2023-01-07 um 08 53 44" src="https://user-images.githubusercontent.com/11976694/211140259-a5ab291e-1cd6-4c9c-9818-096366edcc4f.png">

The relevant field has type "hwaddr" and this is one of these globally defined types in core specs ... (core specs page 391)

Reply:
Thats why I thoight to better have them there? if I'm wrong we can delete the TODO